flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/ExpressionParser.scala

```diff
@@ -45,6 +45,13 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
     ("""(?i)\Q""" + kw.key + """\E""").r
   }
 
+  // Convert the keyword into an case insensitive Parser
+  // It uses an exact matching mode. scenes to be used:
+  // a keyword as a suffix no required parameter and not as a prefix
+  def keyword2ParserForSuffixButNotAsPrefix(kw: Keyword): Parser[String] = {
```

Comment:
I felt like this should be by default applied to all Keywords when using `suffix`

Reply:
probably directly changing the `implicit def keyword2Parser` might have been a better idea?

Reply:
@walterddr I have tried your idea, but some suffix keywords can be used as a prefix at the same time, and some suffix keywords can take parameters. Applying this rule to them will cause their test cases to fail.

Reply:
I dont necessarily think making the suffix keyword to always situated at end of line (`$` assertion) is a must. We just need to prevent potentially keyword collision with identical prefix. Probably adding an assertion that `Keyword` does not follow by alphabetical character is suffice?

Reply:
Yes, I agree with @walterddr. This seems to be a better approach. 
A keyword should only match if it is not followed by another identifier (see [Java identifier specs](https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8)) character. 

What do you think @yanghua?

Reply:
@fhueske @walterddr I agree with you and have updated the PR. Please review again, thank you.